A new file on civilized discussions. (#392)
exercism/docs: community/civilized-discussion.md

<a name="civilized"></a>

## [This is a Civilized Place for Public Discussion](#civilized)

> Please treat this discussion forum with the same respect you would a public park. We, too, are a shared community resource &mdash; a place to share skills, knowledge and interests through ongoing conversation.

Reviewer: > Please treat this discussion forum with the same respect you would a public park.

Given how much trash there is in public parks, I'm not sure this works the way you want it to work. Maybe a public library fits the sentiment better, especially in regards to what follows behind the dash?

Reviewer 2: I agree with Sascha that people really don't seem to respect public parks, at all.

These are not hard and fast rules. They are guidelines to aid the human judgment of our community and keep this a kind, friendly place for civilized public discourse. Please read this document carefully to avoid accidentally violating these guidelines.

If we see posts that we judge to not meet these guidelines, we will actively remove them, but will also let you know privately why we're doing so. 

Everything in this forum is also covered by our [Code of Conduct](https://exercism.org).

<a name="encourage"></a>

> ## [Encourage each other](#improve)

Reviewer 2: ```suggestion
## [Encourage each other](#encourage)
```

> This is first and foremost an educational space - a space to learn and share ideas, to uplift each other. It is **not** a space to prove yourself, show off your knowledge, compete with others, create a sense of “us vs them”, or put others down. We take a strict approach to this, and so topics that would be considered ok in other forums are not considered ok here. 

Reviewer 3: > This is first and foremost an educational space - a space to learn and share ideas, to uplift each other.

Is this meant to cover exercism.org as a platform, or also the GitHub organization (issues, PRs, etc)?

If it also covers GitHub then we might need to reword, because the GitHub org isn't primarily an educational space.

We might also reconsider "topics" and "other forums" in this paragraph.

Author: Good point. I think I'd initially like to use this PR review to get opinions for the context of the forum, so everyone's had a space to discuss that. Then we can update it to go org-wide. I'll update the OP to clarify.

Some examples:

> - A discussion exploring what the tradeoffs would be between using C# vs F# could be a positive and educational experience. However, a comment saying “C# is rubbish because …” is not helpful, does not add to the educational experience and would be deleted.

Reviewer 2: ```suggestion
- A discussion exploring what the trade-offs would be between using C# vs F# could be a positive and educational experience. However, a comment saying “C# is rubbish because …” is not helpful, does not add to the educational experience, and would be deleted.
```

> - Any conversations around topics such as politics or religion are not appropriate in this space. Neither are topics that are fought with a religious fervor (e.g. whether VIM or Emacs is better).

Reviewer 2: This would disallow me to advocate for accessibility, FYI. That may be intentional, or it may not be what you intended, but from the outside, it absolutely would be considered _preaching_ by some. For example, read the incredible toxic and hostile responses here: https://github.com/prettier/prettier/issues/7475.

- Any sort of comment that implies “noobs do …” or “real developers use …” will be deleted.

<a name="improve"></a>

## [Improve the Discussion](#improve)

> Help us make this a great place for discussion by **always adding something positive to the discussion**. If you are not sure your post adds to the conversation, think over what you want to say and try again later.

Reviewer 2: ```suggestion
Help us make this a great place for discussion by **always adding something positive or valuable to the discussion**. If you are not sure your post adds to the conversation, think over what you want to say and try again later.
```

If you're forcing me to do a feedback sandwich, this probably would not be a place for me. Whilst I usually have something positive to say, it's not _always_ positive, and forcing it undermines the value of the feedback itself (apart from it being incredibly rude in my culture to feedback sandwich).

Reviewer 3: @SleeplessByte Your contributions to the JavaScript mutation discussion in the forum today were exemplary. Extremely valuable, and I also saw it as positive. I vehemently dislike feedback sandwiches (it's also seen as rude in the culture I'm from).

One way to improve the discussion is by discovering ones that are already happening. Spend time browsing the topics here before replying or starting your own, and you’ll have a better chance of meeting others who share your interests.

> The topics discussed here matter to us, and we want you to act as if they matter to you, too. Be respectful of the topics and the people discussing them, even if you disagree with some of what is being said.

Reviewer: "adding something **positive** to the discussion" doesn't match the rest of the paragraph imo. A post can be negative, e.g. because it points out an issue with an approach or whatever, but still be valuable and useful for the discussion.

Reviewer 4: What about `always adding something valuable to the discussion`?

Reviewer: I think that's better

<a name="agreeable"></a>

## [Be Agreeable, Even When You Disagree](#agreeable)

> You may wish to respond by disagreeing. That’s fine. But remember to _criticize ideas, not people_. Please avoid:

Reviewer 2: I would like to see stronger language instead of `Please avoid` like: "We do not tolerate:"

* Name-calling
* Ad hominem attacks
* Responding to a post’s tone instead of its actual content
* Knee-jerk contradiction

Instead, provide thoughtful insights that improve the conversation.

> Write your opinions **as opinions**. Rather than saying "everyone knows Python is the best language for beginners" or "Python is the best language for beginners", say "I feel that Python is the best language for beginners because ...". We take this seriously as it's an innocent looking habit that can be extremely disruptive to an online community. If you constantly write opinions as facts, we will ask you to stop and refer you to these guidelines. If you continue, we may choose to ban you.

Reviewer: ```suggestion
Write your opinions **as I-messages**. Rather than saying "everyone knows Python is the best language for beginners" or "Python is the best language for beginners", say "I feel that Python is the best language for beginners because ...". We take this seriously as it's an innocent looking habit that can be extremely disruptive to an online community. If you constantly write opinions as facts, we will ask you to stop and refer you to these guidelines. If you continue, we may choose to ban you.
```

I feel that explicitly refering to I-messages is clearer here because it's not uncommon to assume that everything one writes in a forum is an opinion. Alternatively maybe something like this could underline that such an assumption should not be made:

```suggestion
Be **explicit** about your opinions **being opinions**. Rather than saying "everyone knows Python is the best language for beginners" or "Python is the best language for beginners", say "I feel that Python is the best language for beginners because ...". We take this seriously as it's an innocent looking habit that can be extremely disruptive to an online community. If you constantly write opinions as facts, we will ask you to stop and refer you to these guidelines. If you continue, we may choose to ban you.
```

Author: Thanks. I prefer this second suggestion to the first as I don't know what "I messages are" (I can guess, but others might not guess). @kytrinyx Any thoughts here?

Reviewer: https://en.wikipedia.org/wiki/I-message

Reviewer 4: I've also never heard of an "I-message", so that might be something that people in general don't know about.

Reviewer 3: I hadn't heard of it, either. I don't know if this is something that a lot of people haven't heard of, or it just happens to be me and Erik. I like the wikipedia article on it, though.

If it's likely that a lot of people haven't heard about it, I would lean towards not introducing the idea here, because it means we're potentially leading people away from this article, whereas I really, really want people to read the whole thing once they've started, because it seems less likely that they'll finish if they click away (and we want people to have read it before they start posting in the forum).

Reviewer 2: Whatever we do, I do really like the _extra explicitness_ here that Sascha is trying to add. The second suggestion would be much better than the current doc imo.

Reviewer 5: I also never heard of I-messages before.

<a name="derailing"></a>

> ## [Stay on topic - don’t derail](#derail)

Reviewer 2: ```suggestion
## [Stay on topic - don’t derail](#derailing)
```

When posting, always think about whether you’re moving the conversation in the direction that the original poster intended, or whether you’re pulling it in a different direction. If it’s the latter, it's generally better to create a new topic.

> For example, if there’s a topic on cats, but you prefer dogs, don’t create a post saying “cats are rubbish - I like dogs” - instead create a new topic on dogs with your favorite puppy pictures in.

Reviewer: I'd like to report a violation: http://forum.exercism.org/t/jeremy-prefers-dogs-in-a-post-about-cats/72/2

Author: Yes - this was deliberate :) You'll note I was positive "I don't normally like dogs but...". However, we did raise an internal issue about @lorettaexercism's behaviour in that thread ;)

Let’s leave our community better than we found it.

<a name="flag-problems"></a>

## [If You See a Problem, Flag It](#flag-problems)

Moderators have special authority; they are responsible for this forum. But so are you. With your help, moderators can be community facilitators, not just janitors or police.

When you see bad behavior, don’t reply. Replying encourages bad behavior by acknowledging it, consumes your energy, and wastes everyone’s time. _Just flag it_. If enough flags accrue, action will be taken, either automatically or by moderator intervention.

In order to maintain our community, moderators reserve the right to remove any content and any user account for any reason at any time. Moderators do not preview new posts; the moderators and site operators take no responsibility for any content posted by the community.

<a name="be-civil"></a>

## [Always Be Civil](#be-civil)

Nothing sabotages a healthy conversation like rudeness:

> * Be civil. Don’t post anything that a reasonable person would consider offensive, abusive, or hate speech.

Reviewer: I feel like "reasonable person" is quite judgey here. People will have different ideas of what's offensive, abusive, or hate speech, and implying that someone whose idea of that might be different from the moderators is an "unreasonable person" is quite harsh. This text violates the guidelines above about criticising ideas, not people.

Reviewer 2: I agree with what the point is trying to make and I agree with what Sascha is saying. [Point in case](https://www.npr.org/2020/09/29/917747123/you-literally-cant-believe-the-facts-tucker-carlson-tells-you-so-say-fox-s-lawye) (without a comment about this case, or the outcome, or the subjects, but just take it at face value that "what is reasonable" means very different things to very different people)

> * Keep it clean. Don’t post anything obscene or sexually explicit.

Reviewer: To me using "clean" here feels very judgemental. While sexually explicit content has no place in this forum, it's not "dirty", which this post implies. You can remove the "Keep it clean" and it works just as well without that judgement.

Reviewer 2: It's also not rude to be sexually explicit, but it _is_ rude to be _out of place_. Exercism is not a place to be sexually explicit or obscene.

* Respect each other. Don’t harass or grief anyone, impersonate people, or expose their private information.
* Respect our forum. Don’t post spam or otherwise vandalize the forum.

These are not concrete terms with precise definitions &mdash; avoid even the _appearance_ of any of these things. If you’re unsure, ask yourself how you would feel if your post was featured on the front page of a major news site.

> This is a public forum, and search engines index these discussions. Keep the language, links, and images safe for family and friends.

Reviewer: The idea of what's friendly for family and friends varies _dramatically_ between both cultures and even individuals within those cultures. Think back to the person who thought exercism was a christian minecraft server and considered "making out" obscene language for an example. Perhaps this could avoid that issue:

```suggestion
This is a public forum, and search engines index these discussions. Don't post anything that you wouldn't want to be publicly associated with your name.
```

Reviewer 2: I also have an issue with the wording Safe For Work or "family friendly" content, but I have no issue with _strongly suggesting_ (and heavily enforcing) people don't make the forum inhospitable.

<a name="keep-tidy"></a>

## [Keep It Tidy](#keep-tidy)

Make the effort to put things in the right place, so that we can spend more time discussing and less cleaning up. So:

* Don’t start a topic in the wrong category; please read the category definitions.
* Don’t cross-post the same thing in multiple topics.
* Don’t post no-content replies.

> * Don’t divert a topic by changing it midstream.

Reviewer: Discourse has threads within a thread, kinda like a tree. Sometimes that can lead to interwoven conversations. This rule would effectively disallow using that feature. Is that intended?

* Don’t sign your posts &mdash; every post has your profile information attached to it.

Rather than posting “+1” or “Agreed”, use the Like button. Rather than taking an existing topic in a radically different direction, use Reply as a Linked Topic.

<a name="stealing"></a>

## [Post Only Your Own Stuff](#stealing)

> You may not post anything digital that belongs to someone else without permission. You may not post descriptions of, links to, or methods for stealing someone’s intellectual property (software, video, audio, images), or for breaking any other law.

Reviewer 6: Can we improve the phrasing here? In particular,

> You may not post descriptions of, links to, or methods for [...] breaking any other law.

Any other law, in any country? There are laws that forbid things that we may consider ordinary, moral, and desirable. The set of all comments that does not "break any law" may be much smaller than the set of comments we want to allow.

Edit: I see that Reddit leans heavily on "applicable law" in https://www.redditinc.com/policies/user-agreement. But I'm not sure that's what we want either. And their [content policy](https://www.redditinc.com/policies/content-policy) does have a similar "avoid posting illegal content", but maybe that's defined somewhere as "illegal in your country", or "illegal in the USA".

Reviewer 2: I am not very keen on "you may not break the law" style language. I rather see consequences instead.

Me posting about Tom would be illegal in a lot of countries, so there's that.
